CanHazFunny/CanHazFunny.Tests/JesterTests.cs

```diff
@@ -1,7 +1,91 @@
+using Microsoft.VisualStudio.TestPlatform.TestHost;
+using Moq;
+using System;
 using Xunit;
 
 namespace CanHazFunny.Tests;
 
 public class JesterTests
 {
+    public Mock<IJokeService> JokeServiceMock { get; }
+    public Mock<IJokeOutput> JokeOutputMock { get; }
+    public Jester JesterInstance { get; }
+    public JesterTests()
+    {
+        JokeServiceMock = new Mock<IJokeService>();
+        JokeOutputMock = new Mock<IJokeOutput>();
+        JesterInstance = new Jester(JokeServiceMock.Object, JokeOutputMock.Object);
+    }
+    [Fact]
+    public void JokeOutput_Joke_WriteToConsole()
+    {
+    
+        Mock<IJokeOutput> jokeOutputMock = new();
+        string expectedJoke = "This is a super funny joke!";
+
+        
+        jokeOutputMock.Object.WriteJoke(expectedJoke);
+
+        jokeOutputMock.Verify(jokeoutput => jokeoutput.WriteJoke(expectedJoke), Times.Once);
+    }
+    [Fact]
+    public void JokeService_Joke_Success()
+    {
+        Mock<IJokeService> jokeServiceMock = new();
+        string expectedJoke = "This is a super funny joke!";
+        jokeServiceMock.Setup(jokeservice => jokeservice.GetJoke()).Returns(expectedJoke);
+
+        string result = jokeServiceMock.Object.GetJoke();
+
+        Assert.Equal(expectedJoke, result);
+    }
+    [Fact]
+    public void Constructor_NullJokeService_ThrowsArgumentNullException()
+    {
+        IJokeService nullJokeService = null!;
+        IJokeOutput mockJokeOutput = Mock.Of<IJokeOutput>();
+
+        Assert.Throws<ArgumentNullException>(() => new Jester(nullJokeService, mockJokeOutput));
```

Comment:
```suggestion
        IJokeOutput mockJokeOutput = Mock.Of<IJokeOutput>();

        Assert.Throws<ArgumentNullException>(() => new Jester(null!, mockJokeOutput));
```
This seems simpler and cleaner